tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp

```diff
@@ -2,18 +2,34 @@
 #ifndef LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SCALE_HPP_
 #define LINALG_TPLIMPLEMENTATIONS_INCLUDE_EXPERIMENTAL___P1673_BITS_KOKKOSKERNELS_SCALE_HPP_
 
+#include "signal_kokkos_impl_called.hpp"
+
 namespace KokkosKernelsSTD {
-template<class ExecSpace,
+
+//
+// for now, specialize for default_accessor
+// https://github.com/kokkos/stdBLAS/issues/122
+//
+template<class ExeSpace,
          class Scalar,
          class ElementType,
          std::experimental::extents<>::size_type ... ext,
-         class Layout,
-         class Accessor>
-void scale(kokkos_exec<ExecSpace>, const Scalar alpha,
-           std::experimental::mdspan<ElementType, std::experimental::extents<ext ...>, Layout, Accessor> x)
+         class Layout>
+requires (sizeof...(ext) <= 2)
```

Comment:
This is a perfectly fine way to express the constraint.  You could also add a `requires` clause after the function's parameters, and do `requires(obj.rank() <= 2)`.

Reply:
Just to clarify: Either way is fine with me : - )